sql/src/main/java/io/crate/operation/collect/sources/InformationSchemaIterables.java

```diff
@@ -133,8 +142,18 @@ public InformationSchemaIterables(final Schemas schemas,
         return ingestionRules;
     }
 
-    public Iterable<Void> keyColumnUsageInfos() {
-        return keyColumnUsages;
+    public List<KeyColumnUsage> keyColumnUsage() {
+        return primaryKeyTableInfos.stream()
+            .filter(tableInfo -> !tableInfo.ident().schema().equals(InformationSchemaInfo.NAME))
```

Comment:
I think we should avoid creating an extra list here, it is unnecessary and can easily be avoided ( with `return () -> ...stream()...iterator()`) 

Another way to slightly optimize this might be to have a set with all 3 schema names and then just use 1 filter with a contains check

Reply:
Thanks for the suggestion with the list. I think filter advice is a case for premature optimization.

Reply:
> I think filter advice is a case for premature optimization.

Well, I think it would also be more straightforward way to do that filter anyway =)
Btw, what about `pg_catalog` ?
 I think what's also bugging me about this here is that it's unclear why these 3 schemas are excluded. 
E.g. a `SYS_SCHEMAS.contains..` + expclicit BlobSchema check would convey better why they're excluded.
Would probably be also more resilient towards changes (what if we add another system schema in the future - how would we notice that this needs to be updated)

Reply:
Ok granted, it is more efficient and keeps the logic in place. Changed it. 

>Would probably be also more resilient towards changes (what if we add another system schema in the future - how would we notice that this needs to be updated)

We would notice immediately because there is a test case which would fail due to new results being returned.

`pg_catalog` does not appear here? https://www.postgresql.org/docs/8.4/static/infoschema-key-column-usage.html What do you mean?

Reply:
> pg_catalog does not appear here? 

It's another system schema that we have. But I guess there are no tables with primary keys yet.

So I guess this is good as it is if we have a test that starts failing once we add stuff

Reply:
I've added it, just in case.